pkg/lockfile/lockfile_test.go

```diff
@@ -213,7 +213,9 @@ func TestLockfileName(t *testing.T) {
 
 	assert.NotEmpty(t, l.name, "lockfile name should be recorded correctly")
 
-	l.Lock()
+	for l.TryLock() != nil {
+		time.Sleep(10 * time.Millisecond)
+	}
```

Comment:
Actual tests of the “try” behavior (when locked, and unlocked) would be useful, for both RO and RW.

In new functions, not in a test named `TestLockfileName`,  nor changing what the existing `TestLockfileRead` does. (Dare I ask for 100% code coverage for these primitives?)